### DISCUSSION TITLE
destructure joins?

### DISCUSSION THREAD

**Discussion by halindraprakoso** [prisma/prisma#11999] [Q&A] [answered]
Hello,

say I have this query: 
```js
const data = await db.user.findUnique({
    where: { id: params.id },
    include: { transactions: { orderBy: { createdAt: "asc" } } },
  });
```

instead of having `transactions` inside `data`, can I have `user` and `transactions` separate?
```js
const {user, transactions} = await db.user.findUnique({
    where: { id: params.id },
    include: { transactions: { orderBy: { createdAt: "asc" } } },
  });
```

**Comment by austincrim** [comment, marked as answer]
Hey @halindraprakoso 👋,

There is not a Prisma-specific configuration to achieve this result but you might be able to use the rest operator to achieve something similar.

```ts
const { transactions, ...allOtherFields } = await db.user.findUnique({
  where: { id: params.id },
  include: { transactions: { orderBy: { createdAt: "asc" } } }
});
```

If you have an idea for how we could handle this kind of configuration, I would encourage you to open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with more details!

**Reply by halindraprakoso** [reply]
thanks! I think that's sufficient.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
